includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -718,8 +718,30 @@ public function maybe_process_upe_redirect() {
 				if ( isset( $_GET['redirect_status'] ) && 'succeeded' === $_GET['redirect_status'] ) {
 					$customer = new WC_Stripe_Customer( wp_get_current_user()->ID );
 					$customer->clear_cache();
-
 					wc_add_notice( __( 'Payment method successfully added.', 'woocommerce-gateway-stripe' ) );
+
+					// The newly created payment method does not inherit the customers' billing info, so we manually
+					// trigger an update; in case of failure we log the error and continue because the payment method's
+					// billing info will be updated when the customer makes a purchase anyway.
+					try {
+						$setup_intent_id = isset( $_GET['setup_intent'] ) ? wc_clean( wp_unslash( $_GET['setup_intent'] ) ) : '';
+						$setup_intent = WC_Stripe_API::request( [], 'setup_intents/' . $setup_intent_id, 'GET' );
+
+						$customer_data = WC_Stripe_Customer::map_customer_data( null, new WC_Customer( wp_get_current_user()->ID ) );
+						WC_Stripe_API::request(
+							[
+								'billing_details' => [
+									'name'    => $customer_data['name'],
+									'email'   => $customer_data['email'],
+									'phone'   => $customer_data['phone'],
+									'address' => $customer_data['address'],
+								],
+							],
+							'payment_methods/' . $setup_intent->payment_method
+						);
+					} catch ( Exception $e ) {
```

Comment:
```suggestion
                         do_action( 'woocommerce_stripe_add_payment_method', $user->get_id(), $setup_intent->payment_method );
					} catch ( Exception $e ) {
```
When we add new payment methods during checkout [here](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/payment-methods/class-wc-stripe-upe-payment-gateway.php#L848), we call the `woocommerce_stripe_add_payment_method` hook: should we be doing the same thing here as well? If you decide to do so, you may have to do a little more work, since that final hook parameter is expecting a payment method object; you could expand the `payment_method` field in the request to `/setup_intents/seti_...`, for example...Just food for thought.

Reply:
We can use the response from the `payment_methods/<pm_id>` POST that contains the full payment method object as the hook's final param.

But I'm not sure if we should do this here or if it would be better to add the call when we sync any new payment method from Stripe (when listing the payment methods)...
If we do it here, we need to add some sort of retry to the API call or risk the hook not being called (or called without the payment method object), so maybe the other alternative is the best, what do your think @FangedParakeet?

Reply:
I'm honestly not sure of how this `woocommerce_stripe_add_payment_method` is currently being used by any other plugins, but it makes sense to me to fire the action when the payment method is actually added, rather than when we actually create the WC Payment Token. Our current implementation takes advantage of how we meticulously check Stripe for new payment methods to allow us to resist having to create a new payment token when we save the payment method, but I don't think it's obvous to any end-user that would like to hook into this action that this is how things work--it wasn't obvious to me and I wrote half of that code!

So yeah, I would argue that we should fire the action when we actually add the payment method on Stripe's side, which is also a bit closer to what the hook name `woocommerce_stripe_add_payment_method` suggests it might do. Please let me know if we actually have these hooks documented somewhere and are saying otherwise though!

With that being my vantage point, I'd say that we should fire the action here or [here](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/class-wc-stripe-intent-controller.php#L472-L502), where we create the setup intent. It seems like it would be easier to do it here, since we are making the `POST` to `payment_methods/<pm_id>`, so, as you stated, we do have the payment method object at our disposal. Will we need to add a retry to the API? If the `POST` request succeeds, Stripe guarantees us that it will return the payment method object; if not, I think we can be fairly confident that something has gone awry with saving the payment method. Perhaps you could argue that we'd be more confident of the payment method having been saved when we create the setup intent, but we'd have to make another request or expand the payment method in order to have access to it at that point. 

I think we can safely drop the hook here, but if you think otherwise, I'm happy to yield to your better judgement.

Reply:
I agree, you have a really good point... I added the call to the action hook here 👍🏼